micrometer-core/src/main/java/io/micrometer/core/instrument/step/StepValue.java

```diff
@@ -34,13 +35,18 @@
 
     private final long stepMillis;
 
-    private AtomicLong lastInitPos;
+    private final AtomicLong lastInitPos;
 
-    private volatile V previous = noValue();
+    private volatile V previous;
 
     public StepValue(final Clock clock, final long stepMillis) {
+        this(clock, stepMillis, null);
+    }
+
+    protected StepValue(final Clock clock, final long stepMillis, @Nullable final V initValue) {
```

Comment:
Does this need to be protected? It looks like it could be package-private instead.

Reply:
StepBucketHistogram is part of the distribution package so this needs to be protected to use this constructor.